### ISSUE TITLE
assertion failed on file open

### ISSUE THREAD

**Issue by bmulholland** [lewis6991/gitsigns.nvim#657]
### Description

When opening a file, I get an error "assertion failed!" It would be helpful if the assertion could have a message about what the assertion actually is, then I could investigate on my own.

Assertion that fails is from this recent commit: 
https://github.com/lewis6991/gitsigns.nvim/commit/21ab05c2629ef613b1c3452d12469f936855648b#diff-021c5d7ce28245274d36576b654e671cdb2bf55f5d32642fb844002229b3903fR304

### Neovim version

0.8

### Operating system and version

macOS 13.0

### Actual behavior

<img width="1310" alt="Screenshot 2022-10-25 at 14 47 35" src="https://user-images.githubusercontent.com/247713/197776923-2552a0af-c5f0-44f6-aa7a-2be3d03dbb61.png">

### Minimal config

```Lua
I'll put together a minimal repro if this isn't easy to investigate on its own, let me know if you need it. It's a lot to ask for me to do 30 minutes of work to report any error.
```

### Steps to reproduce

As above

**Comment by lewis6991** [comment]
Did you have git available on your PATH when you hit the error?

The error is simply caused by `git --version` not returning anything.

**Comment by bmulholland** [comment]
Yes it was in my path; I even verified that `git --version` in my terminal output the correct version format.

**Comment by lewis6991** [comment]
Please forgive me but the fact that `brew reinstall git`, and `sudo xcode-select --install` fixed the issue **really** suggests that it wasn't. Or maybe the binary didn't work and needed a reinstall from source or something?

Other than that I have absolutely no idea what could have caused that problem, and you aren't the first person to report it.

**Comment by bmulholland** [comment]
Something weird was definitely going on. It wouldn't be the first time that a macOS update broke some of brew, though. FWIW, it wasn't just gitsigns: telescope broke too, and I fixed it at the same time. I filed https://github.com/nvim-telescope/telescope.nvim/issues/2209 to help others there. But in any case, my system config was stable, and the environment/PATH didn't change at all. I'm experienced enough to diagnose that sort of thing. Like I say, most likely issue was the macOS upgrade breaking brew's git symlink or something.

**Comment by lewis6991** [comment]
Ok. Well I appreciate you looking into this.

We probably need to make the git version check a little more gracious.
